intake/source/csv.py

```diff
@@ -31,23 +32,47 @@ def __init__(self, urlpath, csv_kwargs=None, metadata=None,
             Any parameters that need to be passed to the remote data backend,
             such as credentials.
         """
-        self._urlpath = urlpath
+        self._urlpath = pattern_to_glob(urlpath)
+        self._pattern = urlpath
         self._storage_options = storage_options
         self._csv_kwargs = csv_kwargs or {}
         self._dataframe = None
 
         super(CSVSource, self).__init__(metadata=metadata)
 
+    def _get_filname_args(self, include_path_column):
+        col = include_path_column
```

Comment:
I don't really get what's happening here. Perhaps it would be best to include a description of what can happen in urlpath, and at least one example, the in docstring in init.